managed/services/inventory/services.go

```diff
@@ -434,17 +426,56 @@ func (ss *ServicesService) RemoveCustomLabels(ctx context.Context, req *inventor
 		return nil, errTx
 	}
 
-	// Update scrape configuration
+	if err := ss.updateScrapeConfig(ctx, req.ServiceId); err != nil {
+		return nil, err
+	}
+
+	return &inventorypb.RemoveCustomLabelsResponse{}, nil
+}
+
+// ChangeService updates configuration of a service.
+func (ss *ServicesService) ChangeService(ctx context.Context, req *inventorypb.ChangeServiceRequest) (*inventorypb.ChangeServiceResponse, error) {
+	errTx := ss.db.InTransactionContext(ctx, nil, func(tx *reform.TX) error {
+		service, err := models.FindServiceByID(tx.Querier, req.ServiceId)
+		if err != nil {
+			return err
+		}
+
+		columns := service.ChangeStandardLabels(models.ServiceStandardLabelsParams{
+			Cluster:        req.Cluster,
+			Environment:    req.Environment,
+			ReplicationSet: req.ReplicationSet,
+			ExternalGroup:  req.ExternalGroup,
+		})
+
+		if err = tx.UpdateColumns(service, columns...); err != nil {
+			return err
+		}
+
+		return nil
+	})
+	if errTx != nil {
+		return nil, errTx
+	}
+
+	if err := ss.updateScrapeConfig(ctx, req.ServiceId); err != nil {
+		return nil, err
+	}
+
+	return &inventorypb.ChangeServiceResponse{}, nil
+}
+
```

Comment:
I'm not really understand the approach used in this function.

Why do we use `UpdateColumns` instead of just `Update`? Do we really need to complicate the code?

Why do we use direct orm calls (again `UpdateColumns`) from business-logic level? We can do it from the models level.

Usually we unpacking req structure but in this function we're passing it directly. Can we be consistent here?

Reply:
Agree, I think it should be in the models package to separate model and business logic levels.

Reply:
How would you suggest we address concurrency issues when using `Update` instead of `UpdateColumns`?
In case the service changes between we queried for data and updated it, the database will be in an inconsistent state.

`Update` updates all columns for a record, while `UpdateColumns` just the specific columns we want to touch.

Reply:
Updating only specific columns doesn't solve the problem with inconsistency. We'll end up with the same if several transactions updating the same columns. I suggest to use `Serializable` level for such transactions.

Reply:
The update operation is atomic. Multiple transactions changing the same columns shall always produce a consistent result.
Setting transaction isolation to serializable could help as well. I'm just not sure if we're ready, across the codebase, to retry all failed transactions automatically.
If you think this is feasible, I suggest we create a separate PR for the change.

Reply:
Of course it's up to you but imo we're not solving the problem, just hiding it and all this stuff with specifiyng changed fields is error prone.